DigitalLearningSolutions.Data/Models/BaseSearchableItem.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace DigitalLearningSolutions.Data.Models
+{
+    public abstract class BaseSearchableItem
+    {
+        public string? SearchableNameValue;
+
+        public abstract string SearchableName { get; set; }

```

Comment:
Is there a reason this is not an interface forcing the class to have a get-only SearchableName/SearchString? It seems like we never really use the SearchableNameValue and usually it's just a function of some other class properties that need to be kept in sync with the SearchableNameValue?

Reply:
This needs to be a settable property on BaseSearchableItem for the FuzzySearch query. I needed to be able to set this property while forcing the classes which implement it to implement it's own get when it hasn't been explicitly set and this was the best pattern I could come up with.
Ideally I would want something like 
`public string SearchableName { abstract get; set; }`
but that isn't valid.

Reply:
I see - in that case, could we add a comment about this to avoid confusion above the SearchableNameValue property. And maybe name it SearchableNameOverrideForFuzzySearch or something similar to make it clear that it's what the property does? :)